profiling_router.go

```diff
@@ -0,0 +1,146 @@
+package handlers
+
+import (
+	"context"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"strconv"
+
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/go-utils/logger"
+)
+
+const PprofRoutePrefix = "/debug/pprof"
+
+type profiling struct {
+	Enable bool
+	Auth   authentication
+}
+
+type authentication struct {
+	Username string
+	Password string
+}
+
+func NewProfilingRouter(ctx context.Context) (*Router, error) {
+	log := logger.Get(ctx)
+
+	prof := new(profiling)
+
+	err := prof.initialize()
+	if err != nil {
+		return &Router{}, errgo.Notef(err, "fail to initialize pprof profiling")
+	}
+
+	if !prof.isActivable() {
+		return &Router{}, nil
```

Comment:
Don't you want to return an initialized router in this case?
```suggestion
		return NewRouter(log), nil
```